src/widgets/powered-by/__tests__/powered-by-test.ts

```diff
@@ -0,0 +1,75 @@
+import algoliasearchHelper, { AlgoliaSearchHelper } from 'algoliasearch-helper';
+import { render as preactRender, VNode } from 'preact';
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
+import {
+  createInitOptions,
+  createRenderOptions,
+} from '../../../../test/mock/createWidget';
+import { castToJestMock } from '../../../../test/utils/castToJestMock';
+import { PoweredByProps } from '../../../components/PoweredBy/PoweredBy';
+import poweredBy from '../powered-by';
+
+const render = castToJestMock(preactRender);
+jest.mock('preact', () => {
+  const module = jest.requireActual('preact');
+
+  module.render = jest.fn();
+
+  return module;
+});
+
+describe('poweredBy call', () => {
+  it('throws an exception when no container', () => {
+    expect(poweredBy).toThrowErrorMatchingInlineSnapshot(`
+"The \`container\` option is required.
+
+See documentation: https://www.algolia.com/doc/api-reference/widgets/powered-by/js/"
+`);
+  });
+});
+
+describe('poweredBy', () => {
+  let widget: ReturnType<typeof poweredBy>;
+  let container: string | HTMLElement;
```

Comment:
It's never a `string` in the test, is it?

```suggestion
  let container: HTMLElement;
```